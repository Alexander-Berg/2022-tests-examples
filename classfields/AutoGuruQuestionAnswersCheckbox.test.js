const React = require('react');
const { shallow } = require('enzyme');
const { shallowToJson } = require('enzyme-to-json');

const AutoGuruQuestionAnswersCheckbox = require('./AutoGuruQuestionAnswersCheckbox');
const onAnswerMock = jest.fn();

const ANSWERS = [
    {
        answer: 'первый вариант',
        search_tag: [
            'wide-back-seats',
        ],
        search_tag_exclude: [],
    },
    {
        answer: 'второй вариант',
        body_type_group: [
            '',
        ],
        search_tag: [
            'big-trunk',
        ],
    },
];

let tree;
beforeEach(() => {
    tree = shallow(
        <AutoGuruQuestionAnswersCheckbox
            answers={ ANSWERS }
            onAnswer={ onAnswerMock }
        />,
    );
});

it('должен отрендерить варианты ответа с чекбоксами и кнопкой "Далее"', () => {
    expect(shallowToJson(tree)).toMatchSnapshot();
});

it('должен передать список выбранных ответов при клике по кнопке "Далее"', () => {
    const answerIndex = 1;
    tree.find('CheckboxGroup').simulate('change', [ answerIndex ]);
    tree.find('Button').simulate('click');
    expect(onAnswerMock).toHaveBeenCalledWith([ ANSWERS[answerIndex] ]);
});

it('должен передать пустой список при клике по кнопке "Далее", если ни один ответ не выбран', () => {
    tree.find('Button').simulate('click');
    expect(onAnswerMock).toHaveBeenCalledWith([]);
});
