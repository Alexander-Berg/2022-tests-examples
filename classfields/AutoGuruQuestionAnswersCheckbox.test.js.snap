// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен отрендерить варианты ответа с чекбоксами и кнопкой "Далее" 1`] = `
<div
  className="AutoGuruQuestion__answersCheckbox"
>
  <CheckboxGroup
    className="AutoGuruQuestion__answersCheckboxGroup"
    onChange={[Function]}
    value={Array []}
  >
    <Checkbox
      checked={false}
      className="AutoGuruQuestion__answersItemCheckbox"
      key="0"
      onCheck={[Function]}
      size="m"
      type="checkbox"
      value={0}
    >
      первый вариант
    </Checkbox>
    <Checkbox
      checked={false}
      className="AutoGuruQuestion__answersItemCheckbox"
      key="1"
      onCheck={[Function]}
      size="m"
      type="checkbox"
      value={1}
    >
      второй вариант
    </Checkbox>
  </CheckboxGroup>
  <Button
    className="AutoGuruQuestion__buttonNext"
    color="transparentBlackBlue"
    onClick={[Function]}
    size="l"
    type="button"
    width="narrow"
  >
    Далее
  </Button>
</div>
`;
